ddtrace/contrib/django/patch.py

```diff
@@ -403,58 +405,65 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
             # the middleware chain
             _set_request_tags(django, span, request)
 
-            if response:
-                status = response.status_code
-                span.set_tag("django.response.class", func_name(response))
-                if hasattr(response, "template_name"):
-                    # template_name is a bit of a misnomer, as it could be any of:
-                    # a list of strings, a tuple of strings, a single string, or an instance of Template
-                    # for more detail, see:
-                    # https://docs.djangoproject.com/en/3.0/ref/template-response/#django.template.response.SimpleTemplateResponse.template_name
-                    template = response.template_name
-
-                    if isinstance(template, six.string_types):
-                        template_names = [template]
-                    elif isinstance(
-                        template,
-                        (
-                            list,
-                            tuple,
-                        ),
-                    ):
-                        template_names = template
-                    elif hasattr(template, "template"):
-                        # ^ checking by attribute here because
-                        # django backend implementations don't have a common base
-                        # `.template` is also the most consistent across django versions
-                        template_names = [template.template.name]
+            # The following block can lead to ImproperlyConfigured errors from Django as template
```

Comment:
If this localized to only for `template_names`, we could try/except just that with the specific exception we are expecting and fill out the other tags.

Reply:
From the latest django framework test suite run, it looks like the `template` tags setting does not actually raise any internal django errors - i.e. this has been handled already by moving the exception handling to be inside `_set_requests_tags()` as was done initially. I'll revert this change and leave the larger code block as it was originally.